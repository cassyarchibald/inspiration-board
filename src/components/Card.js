import React, { Component } from "react";
import PropTypes from "prop-types";
import emoji from "emoji-dictionary";

import "./Card.css";

class Card extends Component {
  render() {
    return (
      <section className="card">
        <div className="card__content">
          <section>
            <p className="card__content-text">{this.props.text}</p>
          </section>

          <section>
            {this.props.emoji && (
              <p className="card__content-emoji">
                {emoji.getUnicode(this.props.emoji)
                  ? emoji.getUnicode(this.props.emoji)
                  : this.props.emoji}
              </p>
            )}
          </section>
        </div>
      </section>
    );
  }
}

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  emoji: PropTypes.string
};

export default Card;
