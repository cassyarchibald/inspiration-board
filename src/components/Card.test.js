import React from "react";
import Card from "./Card";
import { shallow } from "enzyme";

describe("Card", () => {
  test("that it matches an existing snapshot", () => {
    const wrapper = shallow(
      <Card id={999} text="Test Card" emoji="heart_eyes" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
