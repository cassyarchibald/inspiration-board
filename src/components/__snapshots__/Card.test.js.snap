// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    emoji="heart_eyes"
    id={999}
    text="Test Card"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card__content"
      >
        <section>
          <p
            className="card__content-text"
          >
            Test Card
          </p>
        </section>
        <section>
          <p
            className="card__content-emoji"
          >
            üòç
          </p>
        </section>
      </div>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section>
            <p
              className="card__content-text"
            >
              Test Card
            </p>
          </section>,
          <section>
            <p
              className="card__content-emoji"
            >
              üòç
            </p>
          </section>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="card__content-text"
            >
              Test Card
            </p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test Card",
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": "Test Card",
            "type": "p",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="card__content-emoji"
            >
              üòç
            </p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "üòç",
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": "üòç",
            "type": "p",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card__content"
        >
          <section>
            <p
              className="card__content-text"
            >
              Test Card
            </p>
          </section>
          <section>
            <p
              className="card__content-emoji"
            >
              üòç
            </p>
          </section>
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section>
              <p
                className="card__content-text"
              >
                Test Card
              </p>
            </section>,
            <section>
              <p
                className="card__content-emoji"
              >
                üòç
              </p>
            </section>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="card__content-text"
              >
                Test Card
              </p>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test Card",
                "className": "card__content-text",
              },
              "ref": null,
              "rendered": "Test Card",
              "type": "p",
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="card__content-emoji"
              >
                üòç
              </p>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "üòç",
                "className": "card__content-emoji",
              },
              "ref": null,
              "rendered": "üòç",
              "type": "p",
            },
            "type": "section",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
